test/snippet/io/alignment_file/alignment_file_output_input_range.cpp

```diff
@@ -0,0 +1,39 @@
+#include <tuple>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/io/alignment_file/all.hpp>
+#include <seqan3/std/filesystem>
+
+int main()
+{
+    auto tmp_dir = std::filesystem::temp_directory_path();
+
+    // Create a /tmp/input.sam for the snippet below
+    {
+        using seqan3::operator""_dna5;
+
+        seqan3::alignment_file_output fout{tmp_dir/"input.sam"};
+
+        std::vector<std::tuple<seqan3::dna5_vector, std::string>> range
+        {
+            { "ACGT"_dna5, "First" },
+            { "NATA"_dna5, "2nd" },
+            { "GATA"_dna5, "Third" }
+        }; // a range of "records"
+
+        fout = range; // will iterate over the records and write them
+    }
+
+    // copying a file in one line:
+    seqan3::alignment_file_output{tmp_dir/"output.sam"} = seqan3::alignment_file_input{tmp_dir/"input.sam"};
+
+    // with seqan3::alignment_file_output as a variable:
+    seqan3::alignment_file_output fout{tmp_dir/"output.sam"};
+    seqan3::alignment_file_input fin{tmp_dir/"input.sam"};
+    fout = fin;
```

Comment:
please use here as well string streams

Reply:
Done, except for the line where it says "copying from file", since I guess there the input has to be a file.